main.go

```diff
@@ -81,5 +96,64 @@ func main() {
 		SignElasticsearchRequests: cfg.SignElasticsearchRequests,
 	}
 
-	svc.Start()
+	svc.Start(ctx)
+}
+
+func configureHealthChecks(ctx context.Context,
+	cfg *config.Config,
+	elasticHTTPClient rchttp.Clienter,
+	producer *kafka.Producer,
+	auditProducer *kafka.Producer,
+	datasetAPIClient *dataset.Client) *healthcheck.HealthCheck {
+
+	hasErrors := false
+
+	versionInfo, err := healthcheck.NewVersionInfo(BuildTime, GitCommit, Version)
+	if err != nil {
+		log.Event(ctx, "error creating version info", log.FATAL, log.Error(err))
+		hasErrors = true
+	}
+
+	hc := healthcheck.New(versionInfo, cfg.HealthCheckCriticalTimeout, cfg.HealthCheckInterval)
+
+	if err = hc.AddCheck("Dataset API", datasetAPIClient.Checker); err != nil {
+		log.Event(ctx, "error creating dataset API health check", log.Error(err))
+		hasErrors = true
+	}
+
+	// zebedee is used only for identity checking
+	zebedeeClient := zebedee.New(cfg.AuthAPIURL)
+	if err = hc.AddCheck("Zebedee", zebedeeClient.Checker); err != nil {
+		log.Event(ctx, "error creating zebedee health check", log.ERROR, log.Error(err))
+		hasErrors = true
+	}
+
+	elasticClient := elastic.NewClientWithHTTPClient(cfg.ElasticSearchAPIURL, cfg.SignElasticsearchRequests, elasticHTTPClient)
```

Comment:
thanks for using this constructor! At least we have a single HTTP elasticsearch client.